### PR TITLE
some warnings on Apple

### PR DESCRIPTION
**Pull request by singalen**
There are more where these came from, but these are the most prominent.

### CODE REVIEW

File: far2l/vtshell.cpp [hunk 1]
```diff
@@ -1019,7 +1019,7 @@ static bool shown_tip_exit = false;
 			return;
 	}
 	
-	~VTShell()
+	virtual ~VTShell()
```
Comment (elfmz) [inline far2l/vtshell.cpp]: but why? its not really needed here - there're no descedants of VTShell, and  if make its d-tor virtual then better to same fot c-tors of interfaces that it derived from

File: far2l/wrap.cpp [hunk 2]
```diff
@@ -3614,7 +3614,7 @@ int WINAPI FarEditorControlA(int Command,void* Param)
 			if (Param)
 			{
 				oldfar::EditorSaveFile *oldsf = (oldfar::EditorSaveFile*) Param;
-				newsf.FileName=(oldsf->FileName)?AnsiToUnicode(oldsf->FileName):nullptr;
+				newsf.FileName=AnsiToUnicode(oldsf->FileName);
```
Comment (elfmz) [inline far2l/wrap.cpp]: this doesnt look as correct at all

Comment (singalen) [reply far2l/wrap.cpp]: So you think these lines should stay the way they are?

Comment (elfmz) [reply far2l/wrap.cpp]: i think better to live without virtual of ~VTShell, or add empty virtual d-tor for all interfaces that it derived from.
Having virtual d-tor of class that derived from bases that have no virtual d-tor 'pushes' programmer to write buggy code, as he could assume that all.

Comment (singalen) [reply far2l/wrap.cpp]: Oh, shame on me, I completely missed the fact that it has base classes. Removing this "fix".

**Pull request merged**